<a href="https://colab.research.google.com/github/grantblairsamson/module-20/blob/main/CreditRisk.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# import the modules
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.metrics import confusion_matrix, classification_report

In [ ]:
lending_data_df = pd.read_csv("lending_data.csv")

lending_data_df.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

In [ ]:
y = lending_data_df["loan_status"]

x = lending_data_df.drop(columns= "loan_status")

In [ ]:
from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test, = train_test_split(x, y, random_state=1)

In [ ]:
from sklearn.linear_model import LogisticRegression

lr_model = LogisticRegression(random_state=1)

lr_model.fit(x_train, y_train)

LogisticRegression(random_state=1)

In [ ]:
y_pred = lr_model.predict(x_test)

In [ ]:
confusion_matrix(y_test, y_pred)

array([[18663,   102],
       [   56,   563]])

In [ ]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      0.99      1.00     18765
           1       0.85      0.91      0.88       619

    accuracy                           0.99     19384
   macro avg       0.92      0.95      0.94     19384
weighted avg       0.99      0.99      0.99     19384



Answer the following question: How well does the logistics regression model predict both the 0 and (healthy loan) and 1 (high-risk loan) labels?

**Answer:** The model does a great job in using the original data to prediciting the values of the healthy loans. Precision was 100% and recall was 99%.

The model is pretty good at predicting the values of high risk loans, but not as good as prediciting the healthy laons. Precision was 85% and recall was 91%.